src/typeInference/binaryEvaluationFunction.ts

```diff
@@ -0,0 +1,6 @@
+import { ValueValue } from '../expressions/dataTypes/Value';
+
+/**
+ * Lambda helper function to the binary operator
+ */
+export type BinaryEvaluationFunction = (left: ValueValue, right: ValueValue) => ValueValue;
```

Comment:
We try to keep the name of files the same as the export, including the capitalization. could you rename the file?